app/pages/sections/order-items/catalogue-solutions/select/routes.js

```diff
@@ -83,17 +92,19 @@ export const catalogueSolutionsSelectRoutes = (authProvider, addContext, session
       const orderItems = sessionManager.getFromSession({ req, key: sessionKeys.orderItems });
       const alreadySelectedItem = orderItems
         .filter((orderItem) => orderItem.catalogueItemId === selectedItem.catalogueItemId);
-      if (alreadySelectedItem.length > 0 && alreadySelectedItem[0].catalogueItemId) {
-        return res.redirect(
-          `${config.baseUrl}/organisation/${orderId}/catalogue-solutions/${alreadySelectedItem[0].catalogueItemId}`,
-        );
-      }
+
       sessionManager.saveToSession({
         req, key: sessionKeys.selectedItemId, value: selectedItemId,
       });
       sessionManager.saveToSession({
         req, key: sessionKeys.selectedItemName, value: selectedItem.name,
       });
+
+      if (alreadySelectedItem.length > 0 && alreadySelectedItem[0].catalogueItemId) {
+        return res.redirect(
+          `${config.baseUrl}/organisation/${orderId}/catalogue-solutions/${alreadySelectedItem[0].catalogueItemId}?solutionAlreadySelected=true`,
+        );
+      }
```

Comment:
Instead of adding a query parameter could we not check to see if the catalogue item already exists in the `router.get('/:orderItemId')` method?

The `orderItemId` name here is misleading as it should actually be the `catalogueItemId` and I'd recommend we rename it isn't too much work.

Reply:
> 
> 
> Instead of adding a query parameter could we not check to see if the catalogue item already exists in the `router.get('/:orderItemId')` method?
 yes will check the other solutions instead of adding query parameter 
> The `orderItemId` name here is misleading as it should actually be the `catalogueItemId` and I'd recommend we rename if it isn't too much work.
ok will check this